components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/certificatemgt/CertificateManagerImpl.java

```diff
@@ -81,6 +145,27 @@ public ResponseCode addCertificateToParentNode(String certificate, String alias,
         }
     }
 
+    @Override
+    public ResponseCode addClientCertificate(APIIdentifier apiIdentifier, String certificate, String alias,
+            String tierName, int tenantId) {
+        ResponseCode responseCode;
+        try {
+            responseCode = certificateMgtUtils.validateCertificate(alias, tenantId, certificate);
+            if (responseCode == ResponseCode.SUCCESS && certificateMgtDAO
+                    .checkWhetherAliasExist(alias, tenantId)) {
+                responseCode = ResponseCode.ALIAS_EXISTS_IN_TRUST_STORE;
+            }
+            if (responseCode == ResponseCode.SUCCESS) {
+                certificateMgtDAO.addClientCertificate(certificate, apiIdentifier, alias, tierName, tenantId);
+            }
```

Comment:
we don't have to check for `responsecode=success` twice if we make line 154 a nested if.  

```
if(success){
   if (exists){
     error
   } esle { //depends on the answer to the above question
   addcert()
  }
} 
```